packages/editor/src/services/EditorHelperState.ts

```diff
@@ -0,0 +1,269 @@
+import { createState, useState } from '@speigg/hookstate'
+
+import { store } from '@xrengine/client-core/src/store'
+import { ClientStorage } from '@xrengine/engine/src/common/classes/ClientStorage'
+import { Engine } from '@xrengine/engine/src/ecs/classes/Engine'
+import { ObjectLayers } from '@xrengine/engine/src/scene/constants/ObjectLayers'
+import {
+  SnapMode,
+  SnapModeType,
+  TransformMode,
+  TransformModeType,
+  TransformPivot,
+  TransformPivotType,
+  TransformSpace
+} from '@xrengine/engine/src/scene/constants/transformConstants'
+
+import { EditorHelperKeys } from '../constants/EditorHelperKeys'
+import { SceneState } from '../functions/sceneRenderFunctions'
+
+type EditorHelperStateType = {
+  isPlayModeEnabled: boolean
+  isFlyModeEnabled: boolean
+  transformMode: TransformModeType
+  transformModeOnCancel: TransformModeType
+  transformPivot: TransformPivotType
+  transformSpace: TransformSpace
+  snapMode: SnapModeType
+  translationSnap: number
+  rotationSnap: number
+  scaleSnap: number
+  gridVisibility: boolean
+  gridHeight: number
+  nodeHelperVisibility: boolean
+  physicsHelperVisibility: boolean
+}
+
+const state = createState<EditorHelperStateType>({
+  isPlayModeEnabled: false,
+  isFlyModeEnabled: false,
+  transformMode: TransformMode.Translate,
+  transformModeOnCancel: TransformMode.Translate,
+  transformSpace: TransformSpace.World,
+  transformPivot: TransformPivot.Selection,
+  snapMode: SnapMode.Grid,
+  translationSnap: 0.5,
+  rotationSnap: 10,
+  scaleSnap: 0.1,
+  gridVisibility: true,
+  gridHeight: 0,
+  nodeHelperVisibility: true,
+  physicsHelperVisibility: true
+})
+
+export async function restoreEditorHelperData(): Promise<void> {
+  if (typeof window !== 'undefined') {
+    const s = {} as EditorHelperStateType
+
+    await Promise.all([
+      ClientStorage.get(EditorHelperKeys.TRANSFORM_MODE).then((v) => {
```

Comment:
will this wait for the setter or just the getter? do we need to wrap this in a promise that resolves when the set promise is fulfilled?

Reply:
This will not be awaited. Since setting the variables will occur only once when loading the engine all the promises will be fulfilled. Same this is happening during the state update in the receptors.